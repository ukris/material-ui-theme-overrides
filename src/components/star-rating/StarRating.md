Non-editable StarRating
```jsx
    <StarRating value={3} />
```