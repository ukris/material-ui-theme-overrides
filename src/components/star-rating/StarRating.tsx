import React, { useEffect } from 'react';
import Rating from '@material-ui/lab/Rating';



export default function SimpleRating(
    {name, value, cellId, updateCellVal } : {name?: string, value: 0 | 1 | 2 | 3 | 4 | 5 | null, cellId?: string, updateCellVal?: (cellId: string, newVal: number | null) => void}) {
  const [val, setVal] = React.useState<number | null>(value);

  useEffect(() => {
    if(val !== value) {
        setVal(value)
    }
  }, [value]); // eslint-disable-line react-hooks/exhaustive-deps
  
  return (
        <Rating
          data-test="table-starrating"
          name={name || 'Rating'}
          value={val}
          onChange={(_event, newValue: number | null) => {
            setVal(newValue);
            (updateCellVal && cellId) && updateCellVal(cellId, newValue)
          }}
        />
  );
}