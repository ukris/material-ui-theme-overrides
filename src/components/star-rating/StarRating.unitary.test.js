import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import StarRating from './StarRating'

configure({ adapter: new Adapter() })


describe('StarRating | Non-editable', () => {
    const values = ['one','two','three']
    const component = (
    <StarRating value={3} />)

    it('renders component correctly', () => {
        const wrapper = mount(component)
        const StarRating = wrapper.find('.MuiRating-root[data-test="table-starrating"]')

        const firstStar = StarRating.props().children[0][0].props.children[1].props.checked
        const secondStar = StarRating.props().children[0][1].props.children[1].props.checked
        const thirdStar = StarRating.props().children[0][2].props.children[1].props.checked
        const fourthStar = StarRating.props().children[0][3].props.children[1].props.checked
        const fifthStar = StarRating.props().children[0][4].props.children[1].props.checked

        expect(StarRating.length).toEqual(1)
        expect(firstStar).toEqual(false)
        expect(secondStar).toEqual(false)
        expect(thirdStar).toEqual(true)
        expect(fourthStar).toEqual(false)
        expect(fifthStar).toEqual(false)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})
