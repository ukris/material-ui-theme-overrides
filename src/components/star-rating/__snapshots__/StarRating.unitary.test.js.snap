// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating | Non-editable matches snapshot 1`] = `
<span
  aria-label={null}
  className="MuiRating-root"
  data-test="table-starrating"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role={null}
>
  <label
    className="MuiRating-label"
    htmlFor="Rating-1"
  >
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
    <span
      className="MuiRating-visuallyhidden"
    >
      1 Star
    </span>
  </label>
  <input
    checked={false}
    className="MuiRating-visuallyhidden"
    disabled={false}
    id="Rating-1"
    name="Rating"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="radio"
    value={1}
  />
  <label
    className="MuiRating-label"
    htmlFor="Rating-2"
  >
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
    <span
      className="MuiRating-visuallyhidden"
    >
      2 Stars
    </span>
  </label>
  <input
    checked={false}
    className="MuiRating-visuallyhidden"
    disabled={false}
    id="Rating-2"
    name="Rating"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="radio"
    value={2}
  />
  <label
    className="MuiRating-label"
    htmlFor="Rating-3"
  >
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
    <span
      className="MuiRating-visuallyhidden"
    >
      3 Stars
    </span>
  </label>
  <input
    checked={true}
    className="MuiRating-visuallyhidden"
    disabled={false}
    id="Rating-3"
    name="Rating"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="radio"
    value={3}
  />
  <label
    className="MuiRating-label"
    htmlFor="Rating-4"
  >
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
    <span
      className="MuiRating-visuallyhidden"
    >
      4 Stars
    </span>
  </label>
  <input
    checked={false}
    className="MuiRating-visuallyhidden"
    disabled={false}
    id="Rating-4"
    name="Rating"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="radio"
    value={4}
  />
  <label
    className="MuiRating-label"
    htmlFor="Rating-5"
  >
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
    <span
      className="MuiRating-visuallyhidden"
    >
      5 Stars
    </span>
  </label>
  <input
    checked={false}
    className="MuiRating-visuallyhidden"
    disabled={false}
    id="Rating-5"
    name="Rating"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="radio"
    value={5}
  />
</span>
`;
